abstractions/typescript/src/serialization/additionalDataHolder.ts

```diff
@@ -0,0 +1,8 @@
+/** Defines a contract for models that can hold additional data besides the described properties. */
+export interface AdditionalDataHolder {
+  /**
+   * Gets the additional data for this object that did not belong to the properties.
+   * @return The additional data for this object.
+   */
+  additionalData: Map<string, unknown>;
```

Comment:
We could merge #1328 here. Using Record instead of Map

Reply:
I'm going to go ahead and merge this PR, this way you can rebase #1328